### DISCUSSION TITLE
Build Error while installing ONNX in Linux

### DISCUSSION THREAD

**Discussion by rahul3161** [onnx/onnx#5416] [Q&A]
In order to install OnnX library for linux after cloning the repository. git clone --recursive https://github.com/Microsoft/onnxruntime

Then ran the build.sh with the following configuration --skip-tests --parallel and --config as Release and also as MinSizeRel

But getting this error:- error: ‘const class google::protobuf::FieldOptions’ has no member named ‘unverified_lazy’

Tried searching the problem but didn't got any particular resolution to this yet availabe. Is there some tag I am missing while executing the build.sh ?

Note:- The protobuf is already updated.

**Comment by jcwchen** [comment]
Hi @rahul3161,
It seems like a build question for onnxruntime: https://github.com/microsoft/onnxruntime.

Did you follow the steps mentioned in https://onnxruntime.ai/docs/build/inferencing.html#linux to build onnxruntime from source on Linux? According to the provided error message, there might be some Protobuf version conflict issues. onnxruntime should have built Protobuf internally.

**Reply by rahul3161** [reply]
Yes I followed this step, I guess the error can be regarding linking with protobuf version.
[ 22%] Linking CXX static library libprotobuf.a
[ 22%] Built target libprotobuf
I am getting the error particularly in this step.

Just a follow up query:-
If I don't install protobuf from source separately, its get installed as per the version needed during the build only?

I tried installing the protobuf version mentioned in the cmkae/deps.txt , it says not found. Although found that in release page of protobuf.

**Reply by jcwchen** [reply]
> If I don't install protobuf from source separately, its get installed as per the version needed during the build only?

Yes. IIRC, onnxruntime will build required Protobuf on the fly when build onnxruntime from source and there is no need to build Protobuf from your own. Have you tried to clean all prebuilt files under onnxruntime directory and then rebuild again? If you still bump into issues, I would suggest you raise this issue in onnxruntime repo to get the best assistance from the runtime experts.

**Reply by rahul3161** [reply]
Yes, I cleared and tried. But still not working. Okay I will post the same issue in the onnxruntime repository.
